src/XeroPHP/Application.php

```diff
@@ -445,4 +450,15 @@ public function delete(Remote\Model $object)
 
         return $object;
     }
+
+    public function requestCompleted($response, $info, $headers) {
+        if($this->requestCompletedCallback instanceof \Closure) {
+            call_user_func($this->requestCompletedCallback, $response, $info, $headers);
+        }
+    }
+
+    public function onRequestCompleted($callback) {
```

Comment:
Maybe this should be called `onRequestSuccess` as `completed` could be interpreted as success AND failed requests.